src/nnue/nnue_feature_transformer.h

```diff
@@ -247,14 +250,22 @@ class FeatureTransformer {
         return !stream.fail();
     }
 
+    // Cast a pointer to a 2 dimensional array of width D
+    template<int D, typename T>
+    static constexpr T (*cast_2D(T* pt))[D] {
+        return (T(*)[D]) pt;
+    }
+
```

Comment:
why is this necessary suddenly

Reply:
The problem is that the small and big accumulators have 2D arrays of different dimensions.  Each access method to those arrays in StateInfo can only have a single return type.  As a result they just return a simple pointer which is then cast to the appropriate sized 2D array for correct indexing.  If you can solve this better I will be very happy.

Reply:
This is now gone.  Thanks!